File: docs/breez-sdk/snippets/csharp/Contacts.cs
using Breez.Sdk.Spark;

namespace BreezSdkSnippets
{
    class Contacts
    {
        async Task AddContact(BreezSdk sdk)
        {
            // ANCHOR: add-contact
            var contact = await sdk.AddContact(request: new AddContactRequest(
                name: "Alice",
                paymentIdentifier: "alice@example.com"
            ));
            Console.WriteLine($"Contact added: {contact}");
            // ANCHOR_END: add-contact
        }

        async Task UpdateContact(BreezSdk sdk)
        {
            // ANCHOR: update-contact
            var contactId = "contact-id";
            var contact = await sdk.UpdateContact(request: new UpdateContactRequest(
                id: contactId,
                name: "Alice Smith",
                paymentIdentifier: "alice.smith@example.com"
            ));
            Console.WriteLine($"Contact updated: {contact}");
            // ANCHOR_END: update-contact
        }

        async Task DeleteContact(BreezSdk sdk)
        {
            // ANCHOR: delete-contact
            var contactId = "contact-id";
            await sdk.DeleteContact(id: contactId);
            Console.WriteLine("Contact deleted");
            // ANCHOR_END: delete-contact
        }

        async Task ListContacts(BreezSdk sdk)
        {
            // ANCHOR: list-contacts
            // List contacts with pagination (e.g., 10 contacts starting from offset 0)
            // Optionally filter by exact name match
            var contacts = await sdk.ListContacts(request: new ListContactsRequest(
                name: null, // Set to "Alice" to filter by name
                offset: 0,
                limit: 10
            ));
            foreach (var contact in contacts)
            {
                Console.WriteLine($"Contact: id={contact.id}, name={contact.name}, identifier={contact.paymentIdentifier}");
            }
            // ANCHOR_END: list-contacts
        }
    }
}

File: docs/breez-sdk/snippets/go/contacts.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-spark-go/breez_sdk_spark"
)

func AddContact(sdk *breez_sdk_spark.BreezSdk) (*breez_sdk_spark.Contact, error) {
	// ANCHOR: add-contact
	contact, err := sdk.AddContact(breez_sdk_spark.AddContactRequest{
		Name:              "Alice",
		PaymentIdentifier: "alice@example.com",
	})
	if err != nil {
		return nil, err
	}

	log.Printf("Contact added: %v", contact)
	// ANCHOR_END: add-contact
	return &contact, nil
}

func UpdateContact(sdk *breez_sdk_spark.BreezSdk) (*breez_sdk_spark.Contact, error) {
	// ANCHOR: update-contact
	contactId := "contact-id"
	contact, err := sdk.UpdateContact(breez_sdk_spark.UpdateContactRequest{
		Id:                contactId,
		Name:              "Alice Smith",
		PaymentIdentifier: "alice.smith@example.com",
	})
	if err != nil {
		return nil, err
	}

	log.Printf("Contact updated: %v", contact)
	// ANCHOR_END: update-contact
	return &contact, nil
}

func DeleteContact(sdk *breez_sdk_spark.BreezSdk) error {
	// ANCHOR: delete-contact
	contactId := "contact-id"
	err := sdk.DeleteContact(contactId)
	if err != nil {
		return err
	}

	log.Printf("Contact deleted")
	// ANCHOR_END: delete-contact
	return nil
}

func ListContacts(sdk *breez_sdk_spark.BreezSdk) ([]breez_sdk_spark.Contact, error) {
	// ANCHOR: list-contacts
	// List contacts with pagination (e.g., 10 contacts starting from offset 0)
	// Optionally filter by exact name match
	offset := uint32(0)
	limit := uint32(10)
	contacts, err := sdk.ListContacts(breez_sdk_spark.ListContactsRequest{
		Name:   nil, // Set to &"Alice" to filter by name
		Offset: &offset,
		Limit:  &limit,
	})
	if err != nil {
		return nil, err
	}

	for _, contact := range contacts {
		log.Printf("Contact: id=%v, name=%v, identifier=%v", contact.Id, contact.Name, contact.PaymentIdentifier)
	}
	// ANCHOR_END: list-contacts
	return contacts, nil
}
